syntax = "proto3";

package SvRpc.Example;

message ApplicationMessages
{
	oneof message
	{
		HelloWorldReq HelloWorldReq = 1;
		HelloWorldRes HelloWorldRes = 2;
		HelloRouterReq HelloRouterReq = 3;
		HelloRouterRes HelloRouterRes = 4;
		GetCounterStreamRequest GetCounterStreamRequest = 5;
		GetCounterStreamResponse GetCounterStreamResponse = 6;
	}
}

message HelloWorldReq
{
	string name = 1;
}

message HelloWorldRes
{
	string message = 1;
}

message HelloRouterReq
{
	string name = 1;
}

message HelloRouterRes
{
	string message = 1;
}

message GetCounterStreamRequest
{
	uint32 start = 1;
}
message GetCounterStreamResponse
{
	uint32 counter = 1;
}